apps/leo_storage/src/leo_storage_handler_del_directory.erl

```diff
@@ -491,44 +493,61 @@ run(?TYPE_DEL_BUCKET, ProcState,_MQId,_Directory, State) when ProcState == ?STAT
 
 run(?TYPE_DEL_BUCKET = Type, ProcState, MQId, Directory,
     #state{manager_nodes = ManagerNodes} = State) when ProcState == ?STATE_ENQUEUING ->
-    case notify_current_state_to_manager(
-           ManagerNodes, Directory, ?del_dir_state_to_atom(ProcState)) of
-        true ->
-            case update_state(#del_dir_state{
-                                 mq_id = MQId,
-                                 directory = Directory,
-                                 type = Type,
-                                 state = ProcState,
-                                 is_notification_successful = true,
-                                 timestamp = leo_date:now()}) of
-                ok ->
-                    void;
-                {error, Cause} ->
-                    ?error("run/5", [{cause, Cause}])
-            end;
-        false ->
-            void
+    case update_state(#del_dir_state{
+                         mq_id = MQId,
+                         directory = Directory,
+                         type = Type,
+                         state = ProcState,
+                         is_notification_successful = false,
+                         timestamp = leo_date:now()}) of
+        ok ->
+            void;
+        {error, Cause} ->
+            ?error("run/5 - ENQUEUING ", [{cause, Cause}])
+    end,
+    notify_current_state_to_manager(
+        ManagerNodes, Directory, ?del_dir_state_to_atom(ProcState)),
```

Comment:
After succeeding in `notify_current_state_to_manager/3`, executing `update_state/1` is better because  `is_notification_successful` is changed unconditionally. Was this done on purpose?